.github/workflows/continuous.yml

```diff
@@ -51,26 +98,38 @@ jobs:
         uses: actions/cache@v2
         with:
           path: ${{ env.CACHE_PATH }}
-          key: ${{ runner.os }}-${{ matrix.config }}-cache-${{ github.sha }}
-          restore-keys: ${{ runner.os }}-${{ matrix.config }}-cache
+          key: ${{ runner.os }}-${{ matrix.config }}-${{ matrix.arch }}-cache-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-${{ matrix.config }}-${{ matrix.arch }}-cache
 
       - name: Prepare ccache
         run: |
           ccache --max-size=1.0G
           ccache -V && ccache --show-stats && ccache --zero-stats
 
-      - name: Configure
+      - name: Configure x86
+        if: matrix.arch == 'x86'
+        run: |
+          mkdir -p build
+          cd build
+          cmake .. \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+            -DCMAKE_BUILD_TYPE=${{ matrix.config }}
+
+      - name: Configure arm64
+        if: matrix.arch == 'arm64'
         run: |
           mkdir -p build
           cd build
           cmake .. \
             -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+            -DCMAKE_OSX_ARCHITECTURES=arm64 \
```

Comment:
You can probably merge this step with the previous with `-DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }}`

Reply:
is the other one correct? I searched the doc for `CMAKE_OSX_ARCHITECTURES` and it doesn't specify the values

Reply:
Not clear from the doc, but the [following arch](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-ios-tvos-or-watchos) should be supported:
`armv7;armv7s;arm64;i386;x86_64`